File: pastey-ios/pastey/pastey/Models/Device.swift
//
//  Device.swift
//  pastey
//
//  Created by Burak Duruk on 2024-06-05.
//

import Foundation

// MARK: - Device
struct Device: Codable, Identifiable {
    let id, userID: Int
    let deviceName, publicKey, createdAt: String

    enum CodingKeys: String, CodingKey {
        case id
        case userID = "user_id"
        case deviceName = "device_name"
        case publicKey = "public_key"
        case createdAt = "created_at"
    }
}

typealias ListDevicesResponse = [Device]

// MARK: - CreateDeviceRequest
struct CreateDeviceRequest: Codable {
    let deviceName, publicKey: String

    enum CodingKeys: String, CodingKey {
        case deviceName = "device_name"
        case publicKey = "public_key"
    }
}

// MARK: - CreateDeviceResponse
struct CreateDeviceResponse: Codable {
    let device: Device
    let sessionID, refreshToken, refreshTokenExpiresAt: String

    enum CodingKeys: String, CodingKey {
        case device
        case sessionID = "session_id"
        case refreshToken = "refresh_token"
        case refreshTokenExpiresAt = "refresh_token_expires_at"
    }
}

File: pastey-wails/backend/models/device.go
package models

import "time"

type CreateDeviceResponse struct {
	Device                Device    `json:"device"`
	SessionID             string    `json:"session_id"`
	RefreshToken          string    `json:"refresh_token"`
	RefreshTokenExpiresAt time.Time `json:"refresh_token_expires_at"`
}

type Device struct {
	ID         int64     `json:"id"`
	UserID     int64     `json:"user_id"`
	DeviceName string    `json:"device_name"`
	PublicKey  string    `json:"public_key"`
	CreatedAt  time.Time `json:"created_at"`
}
